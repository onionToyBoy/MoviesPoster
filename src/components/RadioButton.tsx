import React from 'react';

import styles from './RadioButton.module.css';

export const RadioButton: React.FC<{ label: string, isSelected:boolean, fn:()=>void }> = ({label,isSelected, fn}) => {
	const checkStatus = () => {
		return isSelected
			? `${styles.checkBox} ${styles.selected}`
			: styles.checkBox;
	};

	return (
		<div className={styles.container}>
			<div id={'RadioButton'} className={checkStatus()} onClick={fn}></div>
			<div id={'label'} className={styles.label}>{label}</div>
		</div>
	);
};