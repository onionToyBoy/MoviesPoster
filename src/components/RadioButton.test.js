import React from 'react';
import {RadioButton} from './RadioButton';
import {shallow} from 'enzyme';

describe('RadioButton test', () => {
    const onClick = jest.fn();

	test('Should render RadioButton component correct', () => {
        const props= {label:'Test',isSelected: false, fn: onClick}
		const component = shallow(<RadioButton {...props}/>);
        RadioButton
        const fn = component.find({ id: 'RadioButton' }).prop('onClick');
        const label = component.find({ id: 'label' }).prop('children');
        console.log(component.find({ id: 'RadioButton' }).prop('onClick'))

        expect(fn).toBe(props.fn);
        expect(label).toBe(props.label);
		expect(component).toMatchSnapshot();
    });
});