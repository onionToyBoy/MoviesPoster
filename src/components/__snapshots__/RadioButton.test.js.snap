// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton test Should render RadioButton component correct 1`] = `ShallowWrapper {}`;
